### ISSUE TITLE
Support target subshells

### ISSUE THREAD

**Issue by nbrahms** [semgrep/semgrep#3965]
**Is your feature request related to a problem? Please describe.**
The vast majority of UNIX tools can read a file from a subshell:

e.g.
```sh
semgrep -e 'a' --lang generic <(echo 'a')
```

But Semgrep can't:
```
File not found: /dev/fd/11
ran 1 rules on 0 files: 0 findings
```

A workaround exists when scanning exactly one target:

```sh
cat <($SUBSHELL_PROCESS) | semgrep -e "$PATTERN" --lang "$LANG" -
```

but this won't work for multiple subshell targets

**Describe the solution you'd like**
Semgrep should read from a subshell

This is on 0.65

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-423/support-target-subshells.

**Comment by mjambon** [comment]
This a bug, or rather a collection of bugs, in OCaml code (pfff, ocaml-tree-sitter, semgrep-core).

This shell feature is called ["process substitution"](https://www.gnu.org/software/bash/manual/html_node/Process-Substitution.html). I looked into this the other day and I understand what's going on. I fixed this for one use case, but need to fix it for each different kind of parser we use.

What the shell does with `<(echo foo)` is create a temporary file like `/dev/fd/63`:
```
$ ls -l <(echo)
lr-x------ 1 martin martin 64 Oct  2 00:09 /dev/fd/63 -> 'pipe:[24008178]'
```
See explanations and OCaml code for the [`read_file` implementation in pfff](https://github.com/returntocorp/pfff/blob/174c4b07d5556929f4bb44fbe9e5766da25f6c44/commons/Common.ml#L1208-L1242).
